server/src/main/java/timely/store/MetricAgeOffFilter.java

```diff
@@ -0,0 +1,115 @@
+package timely.store;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.Filter;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.commons.collections4.trie.PatriciaTrie;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetricAgeOffFilter extends Filter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MetricAgeOffFilter.class);
+    public static final String DEFAULT_AGEOFF_KEY = "default";
+    private PatriciaTrie<Long> ageoffs = null;
+    private Long currentTime = 0L;
+    private Long defaultAgeOff = 0L;
+    private ByteBuffer buf = ByteBuffer.allocate(256);
+    private byte[] prevMetricBytes = null;
+    private Long prevAgeOff = null;
+
+    @Override
+    public boolean accept(Key k, Value v) {
+
+        // Metrics table row format is PairLexicoder<>(new StringLexicoder(),
+        // new LongLexicoder());
+        // We want to get the bytes for the metric name from the row
+        buf.clear();
```

Comment:
Your idea is a good one.....compare against prevMetricBytes first before pulling the new metric name out.